src/middlewared/middlewared/pytest/unit/utils/test_nss.py

```diff
@@ -95,3 +95,30 @@ def test__check_group_misses():
         with pytest.raises(KeyError) as ve:
             grp.getgrnam(name)
         assert 'name not found' in str(ve)
+
+
+def test___iter_pwd():
+    py_users = {u.pw_uid: u for u in py_pwd.getpwall()}
+
+    for entry in pwd.iterpw():
```

Comment:
Any way to test 'WINBIND' and/or 'SSS'?
Also test failure path?

Reply:
Not really here since we aren't joined to AD or LDAP in the github runner. I figure this validates basic behavior (and gives immediate feedback if someone breaks the parsers) and the full jenkins run will spot errors with SSSD or Winbind.